In [2]:
import pandas as pd

df = pd.read_csv('data/stroopdata.csv')

Congruent  Incongruent
0      12.079       19.278
1      16.791       18.741
2       9.564       21.214
3       8.630       15.687
4      14.669       22.803
5      12.238       20.878
6      14.692       24.572
7       8.987       17.394
8       9.401       20.762
9      14.480       26.282
10     22.328       24.524
11     15.298       18.644
12     15.073       17.510
13     16.929       20.330
14     18.200       35.255
15     12.130       22.158
16     18.495       25.139
17     10.639       20.429
18     11.344       17.425
19     12.369       34.288
20     12.944       23.894
21     14.233       17.960
22     19.710       22.058
23     16.004       21.157

# Inferential Statistic

This report will investigate the [Stroop Effect](https://en.wikipedia.org/wiki/Stroop_effect), by attempting to answer the questions that were provided below. However, this report will based on the data provided by Udacity, which contains the text color naming duration in the two settings (congruent task *E1* and incongruent task *E2*). 

*1. What is our independent variable? What is our dependent variable?*

As the data contains only the duration of each experiments; thus, the **dependent variable** will be the **each task performance (duration to interpret the color)**. In the other hand, the **independent variable** will be **the tasks (the color of the text)**.

*2. What is an appropriate set of hypotheses for this task? What kind of statistical test do you expect to perform? Justify your choices.*

As for I want to see whether the average duration of *E2* ($\mu_2$) will be larger than average duration of *E1* ($\mu_1$), which indicating that *E2* will need longer time to be interpreted. Thus, I construct the hypotheses:

$$H_0: \mu_2-\mu_1 \leq 0$$
$$H_a: \mu_2-\mu_1 > 0$$

Consequently, I intend to conduct a one-tail *t* test ($\alpha = .05$) as there is no population parameters known in this case. 

*3. Report some descriptive statistics regarding this dataset. Include at least one measure of central tendency and at least one measure of variability.*

Here I load the data:

In [3]:
df

Congruent  Incongruent
0      12.079       19.278
1      16.791       18.741
2       9.564       21.214
3       8.630       15.687
4      14.669       22.803
5      12.238       20.878
6      14.692       24.572
7       8.987       17.394
8       9.401       20.762
9      14.480       26.282
10     22.328       24.524
11     15.298       18.644
12     15.073       17.510
13     16.929       20.330
14     18.200       35.255
15     12.130       22.158
16     18.495       25.139
17     10.639       20.429
18     11.344       17.425
19     12.369       34.288
20     12.944       23.894
21     14.233       17.960
22     19.710       22.058
23     16.004       21.157

Now, I want to add a new column that contain each entries difference ($\mu_2-\mu_1$). I also will provide mean of the difference and its standard deviation.

In [10]:
df['d'] = df['Incongruent'] - df['Congruent']
df

Congruent  Incongruent       d
0      12.079       19.278   7.199
1      16.791       18.741   1.950
2       9.564       21.214  11.650
3       8.630       15.687   7.057
4      14.669       22.803   8.134
5      12.238       20.878   8.640
6      14.692       24.572   9.880
7       8.987       17.394   8.407
8       9.401       20.762  11.361
9      14.480       26.282  11.802
10     22.328       24.524   2.196
11     15.298       18.644   3.346
12     15.073       17.510   2.437
13     16.929       20.330   3.401
14     18.200       35.255  17.055
15     12.130       22.158  10.028
16     18.495       25.139   6.644
17     10.639       20.429   9.790
18     11.344       17.425   6.081
19     12.369       34.288  21.919
20     12.944       23.894  10.950
21     14.233       17.960   3.727
22     19.710       22.058   2.348
23     16.004       21.157   5.153

In [11]:
df['d'].mean()

7.964791666666667

In [12]:
df['d'].std()

4.864826910359056

Thus, based on the data I found the difference average or the point estimate ($\mu_d$) = 7.96 and difference standard deviation ($\sigma_d$) = 4.86. In addition, degree of freedom ($df = n - 1$) = 

4. Provide one or two visualizations that show the distribution of the sample data. Write one or two sentences noting what you observe about the plot or plots.

5. Now, perform the statistical test and report your results. What is your confidence level and your critical statistic value? Do you reject the null hypothesis or fail to reject it? Come to a conclusion in terms of the experiment task. Did the results match up with your expectations?

6. Optional: What do you think is responsible for the effects observed? Can you think of an alternative or similar task that would result in a similar effect? Some research about the problem will be helpful for thinking about these two questions!